velox/functions/sparksql/tests/ArithmeticTest.cpp

```diff
@@ -353,6 +353,21 @@ TEST_F(ArithmeticTest, log1p) {
   EXPECT_TRUE(std::isnan(log1p(kNan).value_or(0)));
 }
 
+TEST_F(ArithmetricTest, expm1) {
+  static const auto expm1 = [&](std::optional<double> a) {
+    return evaluateOnce<double>("expm1(c0)", a);
+  }
+
+  const double kE = std::exp(1);
+
+  EXPECT_EQ(expm1(0), 0);
+  EXPECT_EQ(expm1(1), kE - 1);
+  // As this is only for high accuracy of little number, this is just an example
+  EXPECT_LT(expm1(1e-10), 1.00000000005e-10);
+  EXPECT_EQ(expm1(kInf), kInf);
+  EXPECT_TRUE(std::isnan(expm1(kNan).value_or(0)));
```

Comment:
I find these behaviors defined in the documentation of std::expm1. Would you check if they are aligned with Spark? And maybe add tests and document them.

> If the argument is ±0, it is returned, unmodified.
If the argument is -∞, -1 is returned.
If the argument is +∞, +∞ is returned.
If the argument is NaN, NaN is returned.

Reply:
Right. I've check with https://docs.oracle.com/javase/8/docs/api/java/lang/StrictMath.html#expm1-double-, which is the implementation for the Spark, which align with current behavior.